aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -715,11 +716,11 @@ static Tensor & masked_select_out_impl_cpu(Tensor & result, const Tensor & self,
       .check_all_same_dtype(false)
       .resize_outputs(false)
       .add_output(result_strided)
-      .add_input(_self)
-      .add_input(_mask)
+      .add_input(_self.contiguous())
+      .add_input(_mask.contiguous())
```

Comment:
This would be a no-op for contiguous tensors, but would create a copy for non-contiguous ones, right?

Reply:
And why the same fix does not need to be applied to TensorConfig() created on line 737 for parallel kernel?

Reply:
According to @zou3519 only serial kernel is affected, and I didn't figure out yesterday how to fix the kernel itself. I'll take a look now.

Reply:
I root-caused why its serial kernel only, so now I'm sending discontiguous inputs to parallel kernel that can handle them without making additional copies.